# Alumno : Juarez Quispe, Ricardo Juan
# CURSO : LP-III

<br>

# CARGAR DATOS DEL ARCHIVO FALLECIDOS_SINADEF

### Antes de cargar los datos directamente del archivo "fallecidos_sinadef.csv", se realiza un data cleaning de los datos que no seran necesarios para este ejercicio, seguidamente creamos otro archivo .csv que contenga los datos principales que usaremos.

#### Importamos la libreria pandas

In [1]:
import pandas as pd

#### Cargamos el archivo "fallecidos_data_clean.csv"

In [2]:
df=pd.read_csv("fallecidos_data_clean.csv",  delimiter=";", encoding= 'unicode_escape')
df.head()

Nº TIPO SEGURO       SEXO EDAD TIEMPO EDAD
0   1    IGNORADO  MASCULINO   86        AÑOS
1   2    IGNORADO  MASCULINO   64        AÑOS
2   3    IGNORADO  MASCULINO   42        AÑOS
3   4         SIS  MASCULINO   40        AÑOS
4   5         SIS  MASCULINO   43        AÑOS

#### Verificamos que tipos de datos tenemos en nuestro DataFrame

In [3]:
df.dtypes

Nº              int64
TIPO SEGURO    object
SEXO           object
EDAD           object
TIEMPO EDAD    object
dtype: object

#### Despues de realizar un detenido análisis, podemos observar que nuestra columna 'EDAD' es de tipo object lo cual nos dificulta hallar la media respectiva de las edades.
*Es necesario que los datos de la columna 'EDAD' sea de tipo INT, por eso que se averigua el motivo en que la columna 'EDAD' es de tipo object y tenemos como resultado, que es a causa de la variable 'SIN REGISTRO', afectando y obligando que la columna sea de tipo object.*

In [4]:
df[df.EDAD =='SIN REGISTRO' ]

Nº TIPO SEGURO          SEXO          EDAD   TIEMPO EDAD
4270      4271    IGNORADO     MASCULINO  SIN REGISTRO      IGNORADO
5083      5084    IGNORADO     MASCULINO  SIN REGISTRO      IGNORADO
5633      5634    IGNORADO  SIN REGISTRO  SIN REGISTRO  SIN REGISTRO
6288      6289         SIS     MASCULINO  SIN REGISTRO      IGNORADO
12757    12758     ESSALUD     MASCULINO  SIN REGISTRO      IGNORADO
...        ...         ...           ...           ...           ...
498292  498293     ESSALUD      FEMENINO  SIN REGISTRO      IGNORADO
498293  498294     ESSALUD      FEMENINO  SIN REGISTRO      IGNORADO
498880  498881     ESSALUD     MASCULINO  SIN REGISTRO      IGNORADO
499219  499220         SIS      FEMENINO  SIN REGISTRO      IGNORADO
499347  499348    IGNORADO     MASCULINO  SIN REGISTRO      IGNORADO

[440 rows x 5 columns]

#### Antes de continuar con el cambio de tipo de datos en la columna 'EDAD', es necesario saber si todas las filas del dataframe se encuentran completas sin datos en blanco, debido a que esto puede afectar en el resultado cuando vamos a hallar nuestra media.

In [5]:
df.isnull().any()

Nº             False
TIPO SEGURO    False
SEXO           False
EDAD           False
TIEMPO EDAD    False
dtype: bool

**Concluimos que nuestro dataframe se encuentra libre de datos nulos de manera automatica, pero en vez de obtener datos nulos estas han sido rellenadas con el nombre de "SIN REGISTRO", por lo que procedemos a continuar con la eliminación de las filas que contienen este tipo de variable antes de realizar el cambio de tipo de datos de la columna**

*Tenemos un total de 2200 filas de todo nuestro dataframe con una variable 'SIN REGISTRO'*

In [6]:
df[df.EDAD =='SIN REGISTRO'].count().sum()

2200

### Número de filas totales incluidas las que estan sin registrar la edad de los fallecidos

In [7]:
df.count()

Nº             499726
TIPO SEGURO    499726
SEXO           499726
EDAD           499726
TIEMPO EDAD    499726
dtype: int64

### Eliminando todas las filas de los fallecidos que estan sin registrar la edad

In [8]:
df = df.drop(df[df.EDAD == 'SIN REGISTRO'].index)
df.head()

Nº TIPO SEGURO       SEXO EDAD TIEMPO EDAD
0   1    IGNORADO  MASCULINO   86        AÑOS
1   2    IGNORADO  MASCULINO   64        AÑOS
2   3    IGNORADO  MASCULINO   42        AÑOS
3   4         SIS  MASCULINO   40        AÑOS
4   5         SIS  MASCULINO   43        AÑOS

### Comprobación de que se eliminaron los fallecidos sin registrar la edad

In [9]:
df[df.EDAD =='SIN REGISTRO'].count().sum()

0

**Debido a que eliminamos la variable 'SIN REGISTRAR', que obligaba a la columna 'EDAD' que sea de tipo object, ahora se podra cambiar el tipo de dato a int de la columna para hallar la media.**

In [10]:
df['EDAD']=df['EDAD'].astype(str).astype(int)
df.dtypes

Nº              int64
TIPO SEGURO    object
SEXO           object
EDAD            int32
TIEMPO EDAD    object
dtype: object

### Hallamos todos los datos estadísticos básicos de la columna 'EDAD'

In [11]:
df.describe()

Nº           EDAD
count  499286.000000  499286.000000
mean   249836.338401      66.235715
std    144278.610214      23.588758
min         1.000000       0.000000
25%    124842.250000      55.000000
50%    249849.500000      72.000000
75%    374813.750000      83.000000
max    499726.000000    3017.000000

### Finalmente hallamos la media de las edades de las personas fallecidas

In [12]:
df['EDAD'].mean()

66.23571460044944

### Podemos observar que en la columna 'EDAD' el número máximo es de 3017 y esto se debe que dentro de nuestra columna 'TIEMPO EDAD', existe una variable de tipo HORAS.
**Las edades que estan relacionadas con el tiempo de edad en horas, podemos considerar a gran escala como si fuera 1 año de vida, esto es debido al gran volumen de edades que esta siendo manejado en años.**

In [13]:
#De manera lógica es imposible en estos tiempos que una persona sobrepase los 149 años de edad
df[df.EDAD >= 149].count().sum()

75

In [14]:
df[(df['TIEMPO EDAD']=="HORAS")&(df.EDAD>1)]

Nº TIPO SEGURO       SEXO  EDAD TIEMPO EDAD
55          56         SIS   FEMENINO     2       HORAS
85          86         SIS   FEMENINO     8       HORAS
151        152         SIS   FEMENINO     6       HORAS
169        170     ESSALUD   FEMENINO     7       HORAS
175        176         SIS   FEMENINO    13       HORAS
...        ...         ...        ...   ...         ...
499343  499344         SIS   FEMENINO    37       HORAS
499367  499368         SIS  MASCULINO     8       HORAS
499559  499560         SIS  MASCULINO     5       HORAS
499628  499629         SIS  MASCULINO     7       HORAS
499709  499710         SIS   FEMENINO    77       HORAS

[1974 rows x 5 columns]

### Eliminando todas las filas de los fallecidos que estan con horas dentro de la columna tiempo de edad y sus horas de vida ya que estas equivalen a 1 año únicamente para este análisis de datos

In [15]:
df = df.drop(df[(df['TIEMPO EDAD']=="HORAS")&(df.EDAD>1)].index)
df.head()

Nº TIPO SEGURO       SEXO  EDAD TIEMPO EDAD
0   1    IGNORADO  MASCULINO    86        AÑOS
1   2    IGNORADO  MASCULINO    64        AÑOS
2   3    IGNORADO  MASCULINO    42        AÑOS
3   4         SIS  MASCULINO    40        AÑOS
4   5         SIS  MASCULINO    43        AÑOS

### Comprobación de que se eliminaron los fallecidos con horas de vida

In [16]:
df[(df['TIEMPO EDAD']=="HORAS")&(df.EDAD>1)].count().sum()

0

### De igual manera dentro de nuestra columna 'TIEMPO EDAD', existen datos que no han sido registrados estos se tendran que eliminar.

In [17]:
df = df.drop(df[df['TIEMPO EDAD'] =='SIN REGISTRO'].index)
df.head()

Nº TIPO SEGURO       SEXO  EDAD TIEMPO EDAD
0   1    IGNORADO  MASCULINO    86        AÑOS
1   2    IGNORADO  MASCULINO    64        AÑOS
2   3    IGNORADO  MASCULINO    42        AÑOS
3   4         SIS  MASCULINO    40        AÑOS
4   5         SIS  MASCULINO    43        AÑOS

### Comprobación de que se eliminaron los fallecidos sin registrar su tiempo de edad

In [18]:
df[df['TIEMPO EDAD'] =='SIN REGISTRO' ].count().sum()

0

### Volvemos a realizar una comprobación de nuestros dataset para verificar el número máximo de edad
**La edad máxima fue de 148 años**

In [19]:
df.describe()

Nº           EDAD
count  497309.000000  497309.000000
mean   249659.555234      66.441806
std    144305.827290      23.034602
min         1.000000       0.000000
25%    124586.000000      56.000000
50%    249553.000000      72.000000
75%    374664.000000      83.000000
max    499726.000000    3017.000000

### Media en años según la edad de los fallecidos_sinadef

In [20]:
df['EDAD'].mean()

66.44180579880918

# CONCLUSION:
**Podemos observar que nuestra media de la edad en años, no ha sido gravemente afecteada en este nuevo resultado, solo en cifras decimales debido a que los datos que eliminamos dentro del proceso de data cleaning, fueron unicamente datos que pueden ser remplazados con 1 año de edad al ser esos datos de tipo 'EDAD' en un tiempo de edad basado en horas, y la eliminación de estas filas se debe a la jerarquía que maneja el dataframe con respecto a los años en edad.**